### PR TITLE
WT-13809 Add a background thread group that migrates data from source to destination

### PR DESCRIPTION
**Pull request by ajmorton**
Re-delivery of #11320 as we missed initializing a spinlock. This and changing some deprecated `assertEquals` calls to `assertEqual` are the only changes from the original PR.

### CODE REVIEW

File: src/live_restore/live_restore_server.c [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+    __wt_verbose_debug1(
+      session, WT_VERB_FILEOPS, "%s", "Live restore server: Initializing the work queue");
+
+    *work_count = 0;
+    while ((ret = cursor->next(cursor)) == 0) {
+        const char *uri = NULL;
+        WT_ERR(cursor->get_key(cursor, &uri));
+        if (WT_PREFIX_MATCH(uri, "file:")) {
+            __wt_verbose_debug2(
+              session, WT_VERB_FILEOPS, "Live restore server: Adding an %s to the work queue", uri);
```
Comment (Sean04) [inline src/live_restore/live_restore_server.c:194]:
```suggestion
              session, WT_VERB_FILEOPS, "Live restore server: Adding %s to the work queue", uri);
```

**Review by Sean04 — APPROVED** [review]
lgtm

**Review by zunyiliu — APPROVED** [review]
LGTM.

**Comment by luke-pearson** [comment]
evergreen retry

**Pull request merged**